packages/theme/src/cli/utilities/theme-environment/proxy.ts

```diff
@@ -171,6 +171,11 @@ export async function patchRenderingResponse(
   rawResponse: Response,
   patchCallback?: (html: string) => string | undefined,
 ): Promise<Response> {
+  // 304 (Not Modified) responses should be returned
+  if (rawResponse.status === 304) {
```

Comment:
We shouldn't be patching **any** 3xx actually, so perhaps better to check `>= 300 && <400`